--- a/net/minecraft/world/entity/ambient/Bat.java
+++ b/net/minecraft/world/entity/ambient/Bat.java
@@ -38,7 +_,7 @@
     private static final byte DEFAULT_FLAGS = 0;
     public final AnimationState flyAnimationState = new AnimationState();
     public final AnimationState restAnimationState = new AnimationState();
-    private @Nullable BlockPos targetPosition;
+    public @Nullable BlockPos targetPosition;
 
     public Bat(final EntityType<? extends Bat> type, final Level level) {
         super(type, level);
@@ -59,7 +_,7 @@
     }
 
     @Override
-    protected float getSoundVolume() {
+    public float getSoundVolume() {
         return 0.1F;
     }
 
@@ -74,17 +_,17 @@
     }
 
     @Override
-    protected SoundEvent getHurtSound(final DamageSource source) {
+    public SoundEvent getHurtSound(final DamageSource source) {
         return SoundEvents.BAT_HURT;
     }
 
     @Override
-    protected SoundEvent getDeathSound() {
+    public SoundEvent getDeathSound() {
         return SoundEvents.BAT_DEATH;
     }
 
     @Override
-    public boolean isPushable() {
+    public boolean isCollidable(boolean ignoreClimbing) { // Paper - Climbing should not bypass cramming gamerule
         return false;
     }
 
@@ -138,13 +_,13 @@
                     this.yHeadRot = this.random.nextInt(360);
                 }
 
-                if (level.getNearestPlayer(BAT_RESTING_TARGETING, this) != null) {
+                if (level.getNearestPlayer(BAT_RESTING_TARGETING, this) != null && org.bukkit.craftbukkit.event.CraftEventFactory.handleBatToggleSleepEvent(this, true)) { // CraftBukkit - Call BatToggleSleepEvent
                     this.setResting(false);
                     if (!isSilent) {
                         level.levelEvent(null, LevelEvent.SOUND_BAT_LIFTOFF, pos, 0);
                     }
                 }
-            } else {
+            } else if (org.bukkit.craftbukkit.event.CraftEventFactory.handleBatToggleSleepEvent(this, true)) { // CraftBukkit - Call BatToggleSleepEvent
                 this.setResting(false);
                 if (!isSilent) {
                     level.levelEvent(null, LevelEvent.SOUND_BAT_LIFTOFF, pos, 0);
@@ -175,7 +_,7 @@
             float rotDiff = Mth.wrapDegrees(yRotD - this.getYRot());
             this.zza = 0.5F;
             this.setYRot(this.getYRot() + rotDiff);
-            if (this.random.nextInt(100) == 0 && level.getBlockState(above).isRedstoneConductor(level, above)) {
+            if (this.random.nextInt(100) == 0 && level.getBlockState(above).isRedstoneConductor(level, above) && org.bukkit.craftbukkit.event.CraftEventFactory.handleBatToggleSleepEvent(this, false)) { // CraftBukkit - Call BatToggleSleepEvent
                 this.setResting(true);
             }
         }
@@ -201,7 +_,7 @@
             return false;
         }
 
-        if (this.isResting()) {
+        if (this.isResting() && org.bukkit.craftbukkit.event.CraftEventFactory.handleBatToggleSleepEvent(this, true)) { // CraftBukkit - Call BatToggleSleepEvent
             this.setResting(false);
         }
 
