--- a/net/minecraft/tags/TagEntry.java
+++ b/net/minecraft/tags/TagEntry.java
@@ -20,9 +_,9 @@
     );
     public static final Codec<TagEntry> CODEC = Codec.either(ExtraCodecs.TAG_OR_ELEMENT_ID, FULL_CODEC)
         .xmap(e -> e.map(l -> new TagEntry(l, true), r -> (TagEntry)r), entry -> entry.required ? Either.left(entry.elementOrTag()) : Either.right(entry));
-    private final Identifier id;
-    private final boolean tag;
-    private final boolean required;
+    public final Identifier id;
+    public final boolean tag;
+    public final boolean required;
 
     private TagEntry(final Identifier id, final boolean tag, final boolean required) {
         this.id = id;
