--- a/net/minecraft/world/item/component/ItemContainerContents.java
+++ b/net/minecraft/world/item/component/ItemContainerContents.java
@@ -23,7 +_,7 @@
 
 public final class ItemContainerContents implements TooltipProvider {
     private static final int NO_SLOT = -1;
-    private static final int MAX_SIZE = 256;
+    public static final int MAX_SIZE = 256;
     public static final ItemContainerContents EMPTY = new ItemContainerContents(List.of());
     public static final Codec<ItemContainerContents> CODEC = ItemContainerContents.Slot.CODEC
         .sizeLimitedListOf(256)
@@ -31,8 +_,9 @@
     public static final StreamCodec<RegistryFriendlyByteBuf, ItemContainerContents> STREAM_CODEC = ItemStackTemplate.STREAM_CODEC
         .apply(ByteBufCodecs::optional)
         .<List<Optional<ItemStackTemplate>>>apply(ByteBufCodecs.list(256))
+        .apply(ByteBufCodecs::increaseDepth) // Paper - Track codec depth
         .map(ItemContainerContents::new, c -> c.items);
-    private final List<Optional<ItemStackTemplate>> items;
+    public final List<Optional<ItemStackTemplate>> items;
     private final int hashCode;
 
     private ItemContainerContents(final List<Optional<ItemStackTemplate>> items) {
