--- a/paper-api/build.gradle.kts
+++ b/paper-api/build.gradle.kts
@@ -248,3 +_,20 @@
 tasks.check {
     dependsOn(scanJarForBadCalls)
 }
+// Gale start - hide irrelevant compilation warnings
+tasks.withType<JavaCompile> {
+    val compilerArgs = options.compilerArgs
+    compilerArgs.add("-Xlint:-module")
+    compilerArgs.add("-Xlint:-removal")
+    compilerArgs.add("-Xlint:-dep-ann")
+    compilerArgs.add("--add-modules=jdk.incubator.vector")
+}
+// Gale end - hide irrelevant compilation warnings
+tasks.jar {
+    from("${project.projectDir}/LICENSE.txt") {
+        into("")
+    }
+}
