--- a/net/minecraft/world/entity/animal/equine/SkeletonHorse.java
+++ b/net/minecraft/world/entity/animal/equine/SkeletonHorse.java
@@ -38,7 +_,7 @@
         .withAttachments(EntityAttachments.builder().attach(EntityAttachment.PASSENGER, 0.0F, EntityTypes.SKELETON_HORSE.getHeight() - 0.25F, 0.0F))
         .scale(0.7F);
     private boolean isTrap = false;
-    private int trapTime = 0;
+    public int trapTime = 0;
 
     public SkeletonHorse(final EntityType<? extends SkeletonHorse> type, final Level level) {
         super(type, level);
@@ -66,22 +_,22 @@
     }
 
     @Override
-    protected SoundEvent getAmbientSound() {
+    public SoundEvent getAmbientSound() {
         return this.isEyeInFluid(FluidTags.WATER) ? SoundEvents.SKELETON_HORSE_AMBIENT_WATER : SoundEvents.SKELETON_HORSE_AMBIENT;
     }
 
     @Override
-    protected SoundEvent getDeathSound() {
+    public SoundEvent getDeathSound() {
         return SoundEvents.SKELETON_HORSE_DEATH;
     }
 
     @Override
-    protected SoundEvent getHurtSound(final DamageSource source) {
+    public SoundEvent getHurtSound(final DamageSource source) {
         return SoundEvents.SKELETON_HORSE_HURT;
     }
 
     @Override
-    protected SoundEvent getSwimSound() {
+    public SoundEvent getSwimSound() {
         if (this.onGround()) {
             if (!this.isVehicle()) {
                 return SoundEvents.SKELETON_HORSE_STEP_WATER;
@@ -127,7 +_,7 @@
     public void aiStep() {
         super.aiStep();
         if (!this.isPersistenceRequired() && this.isTrap() && this.trapTime++ >= 18000) {
-            this.discard();
+            this.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.DESPAWN); // CraftBukkit - add Bukkit remove cause
         }
     }
 
