--- a/net/minecraft/world/entity/monster/warden/WardenSpawnTracker.java
+++ b/net/minecraft/world/entity/monster/warden/WardenSpawnTracker.java
@@ -28,9 +_,9 @@
     private static final int WARNING_CHECK_DIAMETER = 48;
     private static final int DECREASE_WARNING_LEVEL_EVERY_INTERVAL = 12000;
     private static final int WARNING_LEVEL_INCREASE_COOLDOWN = 200;
-    private int ticksSinceLastWarning;
+    public int ticksSinceLastWarning;
     private int warningLevel;
-    private int cooldownTicks;
+    public int cooldownTicks;
 
     public WardenSpawnTracker(final int ticksSinceLastWarning, final int warningLevel, final int cooldownTicks) {
         this.ticksSinceLastWarning = ticksSinceLastWarning;
@@ -102,7 +_,7 @@
         return level.getPlayers(player -> !player.isSpectator() && player.position().closerThan(origin, 16.0) && player.isAlive());
     }
 
-    private void increaseWarningLevel() {
+    public void increaseWarningLevel() {
         if (!this.onCooldown()) {
             this.ticksSinceLastWarning = 0;
             this.cooldownTicks = 200;
