--- a/net/minecraft/server/level/ServerLevel.java
+++ b/net/minecraft/server/level/ServerLevel.java
@@ -140,3 +_,4 @@
 import net.minecraft.world.level.entity.EntityTickList;
+import org.galemc.gale.entity.EntityPushGrid; // Gale - entity push spatial grid
 import net.minecraft.world.level.entity.EntityTypeTest;
 import net.minecraft.world.level.entity.LevelCallback;
@@ -879,2 +_,9 @@
 
+                                // Gale start - entity push spatial grid
+                                if (this.galeConfig().smallOptimizations.entityPushSpatialGrid && this.galePushGridTick != this.getGameTime()) {
+                                    this.galePushGrid = new EntityPushGrid();
+                                    this.entityTickList.forEach(this.galePushGrid::register);
+                                    this.galePushGridTick = this.getGameTime();
+                                }
+                                // Gale end - entity push spatial grid
                                 entity.stopRiding();
@@ -946,1 +_,3 @@
     private int currentIceAndSnowTick = 0; protected void resetIceAndSnowTick() { this.currentIceAndSnowTick = this.randomTickRandom.nextInt(16); } // Gale - Airplane - optimize random calls in chunk ticking
+    public EntityPushGrid galePushGrid; // Gale - entity push spatial grid
+    public long galePushGridTick = -1; // Gale - entity push spatial grid
