--- a/net/minecraft/world/entity/animal/cow/Cow.java
+++ b/net/minecraft/world/entity/animal/cow/Cow.java
@@ -98,11 +_,11 @@
         return this.entityData.get(DATA_VARIANT_ID);
     }
 
-    private Holder<CowSoundVariant> getSoundVariant() {
+    public Holder<CowSoundVariant> getSoundVariant() {
         return this.entityData.get(DATA_SOUND_VARIANT_ID);
     }
 
-    private void setSoundVariant(final Holder<CowSoundVariant> soundVariant) {
+    public void setSoundVariant(final Holder<CowSoundVariant> soundVariant) {
         this.entityData.set(DATA_SOUND_VARIANT_ID, soundVariant);
     }
 
