--- a/net/minecraft/world/entity/animal/camel/CamelHusk.java
+++ b/net/minecraft/world/entity/animal/camel/CamelHusk.java
@@ -53,7 +_,7 @@
     }
 
     @Override
-    protected SoundEvent getAmbientSound() {
+    public SoundEvent getAmbientSound() {
         return SoundEvents.CAMEL_HUSK_AMBIENT;
     }
 
@@ -78,12 +_,12 @@
     }
 
     @Override
-    protected SoundEvent getDeathSound() {
+    public SoundEvent getDeathSound() {
         return SoundEvents.CAMEL_HUSK_DEATH;
     }
 
     @Override
-    protected SoundEvent getHurtSound(final DamageSource source) {
+    public SoundEvent getHurtSound(final DamageSource source) {
         return SoundEvents.CAMEL_HUSK_HURT;
     }
 
