--- a/net/minecraft/world/level/block/entity/SkullBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/SkullBlockEntity.java
@@ -22,11 +_,11 @@
     private static final String TAG_PROFILE = "profile";
     private static final String TAG_NOTE_BLOCK_SOUND = "note_block_sound";
     private static final String TAG_CUSTOM_NAME = "custom_name";
-    private @Nullable ResolvableProfile owner;
-    private @Nullable Identifier noteBlockSound;
+    public @Nullable ResolvableProfile owner;
+    public @Nullable Identifier noteBlockSound;
     private int animationTickCount;
     private boolean isAnimating;
-    private @Nullable Component customName;
+    public @Nullable Component customName;
 
     public SkullBlockEntity(final BlockPos worldPosition, final BlockState blockState) {
         super(BlockEntityTypes.SKULL, worldPosition, blockState);
