--- a/net/minecraft/world/entity/animal/turtle/Turtle.java
+++ b/net/minecraft/world/entity/animal/turtle/Turtle.java
@@ -77,7 +_,7 @@
     public static final TargetingConditions.Selector BABY_ON_LAND_SELECTOR = (target, level) -> target.isBaby() && !target.isInWater();
     private BlockPos homePos = BlockPos.ZERO;
     private @Nullable BlockPos travelPos;
-    private boolean goingHome;
+    public boolean goingHome;
 
     public Turtle(final EntityType<? extends Turtle> type, final Level level) {
         super(type, level);
@@ -100,7 +_,7 @@
         return this.entityData.get(HAS_EGG);
     }
 
-    private void setHasEgg(final boolean onOff) {
+    public void setHasEgg(final boolean onOff) {
         this.entityData.set(HAS_EGG, onOff);
     }
 
@@ -176,7 +_,7 @@
     }
 
     @Override
-    protected @Nullable SoundEvent getAmbientSound() {
+    public @Nullable SoundEvent getAmbientSound() {
         return !this.isInWater() && this.onGround() && !this.isBaby() ? SoundEvents.TURTLE_AMBIENT_LAND : super.getAmbientSound();
     }
 
@@ -186,17 +_,17 @@
     }
 
     @Override
-    protected SoundEvent getSwimSound() {
+    public SoundEvent getSwimSound() {
         return SoundEvents.TURTLE_SWIM;
     }
 
     @Override
-    protected @Nullable SoundEvent getHurtSound(final DamageSource source) {
+    public @Nullable SoundEvent getHurtSound(final DamageSource source) {
         return this.isBaby() ? SoundEvents.TURTLE_HURT_BABY : SoundEvents.TURTLE_HURT;
     }
 
     @Override
-    protected @Nullable SoundEvent getDeathSound() {
+    public @Nullable SoundEvent getDeathSound() {
         return this.isBaby() ? SoundEvents.TURTLE_DEATH_BABY : SoundEvents.TURTLE_DEATH;
     }
 
@@ -282,7 +_,7 @@
 
     @Override
     public void thunderHit(final ServerLevel level, final LightningBolt lightningBolt) {
-        this.hurtServer(level, this.damageSources().lightningBolt(), Float.MAX_VALUE);
+        this.hurtServer(level, this.damageSources().lightningBolt().eventEntityDamager(lightningBolt), Float.MAX_VALUE); // CraftBukkit // Paper - fix DamageSource API
     }
 
     @Override
@@ -309,6 +_,10 @@
             if (loveCause == null && this.partner.getLoveCause() != null) {
                 loveCause = this.partner.getLoveCause();
             }
+            // Paper start - Add EntityFertilizeEggEvent event
+            io.papermc.paper.event.entity.EntityFertilizeEggEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callEntityFertilizeEggEvent(this.animal, this.partner);
+            if (event.isCancelled()) return;
+            // Paper end - Add EntityFertilizeEggEvent event
 
             if (loveCause != null) {
                 loveCause.awardStat(Stats.ANIMALS_BRED);
@@ -322,7 +_,7 @@
             this.partner.resetLove();
             RandomSource random = this.animal.getRandom();
             if (getServerLevel(this.level).getGameRules().get(GameRules.MOB_DROPS)) {
-                this.level.addFreshEntity(new ExperienceOrb(this.level, this.animal.getX(), this.animal.getY(), this.animal.getZ(), random.nextInt(7) + 1));
+                if (event.getExperience() > 0) this.level.addFreshEntity(new ExperienceOrb(this.level, this.animal.position(), Vec3.ZERO, event.getExperience(), org.bukkit.entity.ExperienceOrb.SpawnReason.BREED, loveCause, this.turtle)); // Paper - Add EntityFertilizeEggEvent event
             }
         }
     }
@@ -345,7 +_,7 @@
                 && (
                     this.turtle.hasEgg()
                         || this.turtle.getRandom().nextInt(reducedTickDelay(700)) == 0 && !this.turtle.homePos.closerToCenterThan(this.turtle.position(), 64.0)
-                );
+                ) && new com.destroystokyo.paper.event.entity.TurtleGoHomeEvent((org.bukkit.entity.Turtle) this.turtle.getBukkitEntity()).callEvent(); // Paper - Turtle API
         }
 
         @Override
@@ -453,14 +_,20 @@
             BlockPos turtlePos = this.turtle.blockPosition();
             if (!this.turtle.isInWater() && this.isReachedTarget()) {
                 if (this.turtle.layEggCounter < 1) {
-                    this.turtle.setLayingEgg(true);
+                    this.turtle.setLayingEgg(new com.destroystokyo.paper.event.entity.TurtleStartDiggingEvent((org.bukkit.entity.Turtle) this.turtle.getBukkitEntity(), org.bukkit.craftbukkit.util.CraftLocation.toBukkit(this.blockPos, this.turtle.level())).callEvent()); // Paper - Turtle API
                 } else if (this.turtle.layEggCounter > this.adjustedTickDelay(200)) {
+                    // Paper start - Turtle API
+                    int eggCount = this.turtle.random.nextInt(4) + 1;
+                    com.destroystokyo.paper.event.entity.TurtleLayEggEvent layEggEvent = new com.destroystokyo.paper.event.entity.TurtleLayEggEvent((org.bukkit.entity.Turtle) this.turtle.getBukkitEntity(), org.bukkit.craftbukkit.util.CraftLocation.toBukkit(this.blockPos.above(), this.turtle.level()), eggCount);
+                    if (layEggEvent.callEvent() && org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(this.turtle, this.blockPos.above(), Blocks.TURTLE_EGG.defaultBlockState().setValue(TurtleEggBlock.EGGS, layEggEvent.getEggCount()))) {
+                    // Paper end - Turtle API
                     Level level = this.turtle.level();
                     level.playSound(null, turtlePos, SoundEvents.TURTLE_LAY_EGG, SoundSource.BLOCKS, 0.3F, 0.9F + level.getRandom().nextFloat() * 0.2F);
                     BlockPos eggPos = this.blockPos.above();
-                    BlockState eggState = Blocks.TURTLE_EGG.defaultBlockState().setValue(TurtleEggBlock.EGGS, this.turtle.random.nextInt(4) + 1);
+                    BlockState eggState = Blocks.TURTLE_EGG.defaultBlockState().setValue(TurtleEggBlock.EGGS, layEggEvent.getEggCount()); // Paper
                     level.setBlock(eggPos, eggState, Block.UPDATE_ALL);
                     level.gameEvent(GameEvent.BLOCK_PLACE, eggPos, GameEvent.Context.of(this.turtle, eggState));
+                    } // CraftBukkit
                     this.turtle.setHasEgg(false);
                     this.turtle.setLayingEgg(false);
                     this.turtle.setInLoveTime(600);
