--- a/net/minecraft/world/entity/monster/Guardian.java
+++ b/net/minecraft/world/entity/monster/Guardian.java
@@ -57,7 +_,8 @@
     private @Nullable LivingEntity clientSideCachedAttackTarget;
     private int clientSideAttackTime;
     private boolean clientSideTouchedGround;
-    protected @Nullable RandomStrollGoal randomStrollGoal;
+    public @Nullable RandomStrollGoal randomStrollGoal;
+    public @Nullable GuardianAttackGoal guardianAttackGoal; // CraftBukkit - add field
 
     public Guardian(final EntityType<? extends Guardian> type, final Level level) {
         super(type, level);
@@ -72,7 +_,7 @@
     protected void registerGoals() {
         MoveTowardsRestrictionGoal goal = new MoveTowardsRestrictionGoal(this, 1.0);
         this.randomStrollGoal = new RandomStrollGoal(this, 1.0, 80);
-        this.goalSelector.addGoal(4, new Guardian.GuardianAttackGoal(this));
+        this.goalSelector.addGoal(4, this.guardianAttackGoal = new Guardian.GuardianAttackGoal(this)); // CraftBukkit - assign field
         this.goalSelector.addGoal(5, goal);
         this.goalSelector.addGoal(7, this.randomStrollGoal);
         this.goalSelector.addGoal(8, new LookAtPlayerGoal(this, Player.class, 8.0F));
@@ -111,7 +_,7 @@
         return 80;
     }
 
-    private void setActiveAttackTarget(final int entityId) {
+    public void setActiveAttackTarget(final int entityId) {
         this.entityData.set(DATA_ID_ATTACK_TARGET, entityId);
     }
 
@@ -153,17 +_,17 @@
     }
 
     @Override
-    protected SoundEvent getAmbientSound() {
+    public SoundEvent getAmbientSound() {
         return this.isInWater() ? SoundEvents.GUARDIAN_AMBIENT : SoundEvents.GUARDIAN_AMBIENT_LAND;
     }
 
     @Override
-    protected SoundEvent getHurtSound(final DamageSource source) {
+    public SoundEvent getHurtSound(final DamageSource source) {
         return this.isInWater() ? SoundEvents.GUARDIAN_HURT : SoundEvents.GUARDIAN_HURT_LAND;
     }
 
     @Override
-    protected SoundEvent getDeathSound() {
+    public SoundEvent getDeathSound() {
         return this.isInWater() ? SoundEvents.GUARDIAN_DEATH : SoundEvents.GUARDIAN_DEATH_LAND;
     }
 
@@ -343,9 +_,9 @@
         }
     }
 
-    private static class GuardianAttackGoal extends Goal {
+    public static class GuardianAttackGoal extends Goal {
         private final Guardian guardian;
-        private int attackTime;
+        public int attackTime;
         private final boolean elder;
 
         public GuardianAttackGoal(final Guardian guardian) {
