--- a/net/minecraft/world/entity/monster/Shulker.java
+++ b/net/minecraft/world/entity/monster/Shulker.java
@@ -69,7 +_,7 @@
     );
     protected static final EntityDataAccessor<Direction> DATA_ATTACH_FACE_ID = SynchedEntityData.defineId(Shulker.class, EntityDataSerializers.DIRECTION);
     protected static final EntityDataAccessor<Byte> DATA_PEEK_ID = SynchedEntityData.defineId(Shulker.class, EntityDataSerializers.BYTE);
-    protected static final EntityDataAccessor<Byte> DATA_COLOR_ID = SynchedEntityData.defineId(Shulker.class, EntityDataSerializers.BYTE);
+    public static final EntityDataAccessor<Byte> DATA_COLOR_ID = SynchedEntityData.defineId(Shulker.class, EntityDataSerializers.BYTE);
     private static final int TELEPORT_STEPS = 6;
     private static final byte NO_COLOR = 16;
     private static final byte DEFAULT_COLOR = 16;
@@ -115,7 +_,7 @@
     }
 
     @Override
-    protected SoundEvent getAmbientSound() {
+    public SoundEvent getAmbientSound() {
         return SoundEvents.SHULKER_AMBIENT;
     }
 
@@ -127,12 +_,12 @@
     }
 
     @Override
-    protected SoundEvent getDeathSound() {
+    public SoundEvent getDeathSound() {
         return SoundEvents.SHULKER_DEATH;
     }
 
     @Override
-    protected SoundEvent getHurtSound(final DamageSource source) {
+    public SoundEvent getHurtSound(final DamageSource source) {
         return this.isClosed() ? SoundEvents.SHULKER_HURT_CLOSED : SoundEvents.SHULKER_HURT;
     }
 
@@ -288,8 +_,10 @@
     }
 
     @Override
-    public void stopRiding() {
-        super.stopRiding();
+    // Paper start - Force entity dismount during teleportation
+    public void stopRiding(boolean suppressCancellation) {
+        super.stopRiding(suppressCancellation);
+        // Paper end - Force entity dismount during teleportation
         if (this.level().isClientSide()) {
             this.clientOldAttachPosition = this.blockPosition();
         }
@@ -400,6 +_,14 @@
                     && this.level().getWorldBorder().isWithinBounds(target)
                     && this.level().noCollision(this, new AABB(target).deflate(1.0E-6))) {
                     Direction attachmentDirection = this.findAttachableSurface(target);
+                    // CraftBukkit start
+                    org.bukkit.event.entity.EntityTeleportEvent teleportEvent = org.bukkit.craftbukkit.event.CraftEventFactory.callEntityTeleportEvent(this, target.getX(), target.getY(), target.getZ());
+                    if (teleportEvent.isCancelled() || teleportEvent.getTo() == null) { // Paper
+                        return false;
+                    } else {
+                        target = org.bukkit.craftbukkit.util.CraftLocation.toBlockPos(teleportEvent.getTo());
+                    }
+                    // CraftBukkit end
                     if (attachmentDirection != null) {
                         this.unRide();
                         this.setAttachFace(attachmentDirection);
@@ -464,7 +_,12 @@
                 if (baby != null) {
                     baby.setVariant(this.getVariant());
                     baby.snapTo(oldPosition);
-                    this.level().addFreshEntity(baby);
+                    // Paper start - Call ShulkerDuplicateEvent
+                    if (!new io.papermc.paper.event.entity.ShulkerDuplicateEvent((org.bukkit.entity.Shulker) baby.getBukkitEntity(), (org.bukkit.entity.Shulker) this.getBukkitEntity()).callEvent()) {
+                        return;
+                    }
+                    // Paper end - Call ShulkerDuplicateEvent
+                    this.level().addFreshEntity(baby, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.BREEDING); // CraftBukkit - the mysteries of life
                 }
             }
         }
@@ -479,7 +_,7 @@
         return this.entityData.get(DATA_ATTACH_FACE_ID);
     }
 
-    private void setAttachFace(final Direction attachmentDirection) {
+    public void setAttachFace(final Direction attachmentDirection) {
         this.entityData.set(DATA_ATTACH_FACE_ID, attachmentDirection);
     }
 
@@ -492,11 +_,11 @@
         super.onSyncedDataUpdated(accessor);
     }
 
-    private int getRawPeekAmount() {
+    public int getRawPeekAmount() {
         return this.entityData.get(DATA_PEEK_ID);
     }
 
-    private void setRawPeekAmount(final int amount) {
+    public void setRawPeekAmount(final int amount) {
         if (!this.level().isClientSide()) {
             this.getAttribute(Attributes.ARMOR).removeModifier(COVERED_ARMOR_MODIFIER_ID);
             if (amount == 0) {
