--- a/net/minecraft/world/entity/monster/Strider.java
+++ b/net/minecraft/world/entity/monster/Strider.java
@@ -87,7 +_,7 @@
     private static final EntityDimensions BABY_DIMENSIONS = EntityDimensions.scalable(0.45F, 0.85F)
         .withEyeHeight(0.4375F)
         .withAttachments(EntityAttachments.builder().attach(EntityAttachment.PASSENGER, 0.0F, 0.65625F, 0.0F));
-    private final ItemBasedSteering steering = new ItemBasedSteering(this.entityData, DATA_BOOST_TIME);
+    public final ItemBasedSteering steering = new ItemBasedSteering(this.entityData, DATA_BOOST_TIME);
     private @Nullable TemptGoal temptGoal;
 
     public Strider(final EntityType<? extends Strider> strider, final Level level) {
@@ -310,7 +_,14 @@
                 || stateOn.is(BlockTags.STRIDER_WARM_BLOCKS)
                 || this.getFluidHeight(FluidTags.LAVA) > 0.0;
             boolean onWarmStrider = this.getVehicle() instanceof Strider strider && !strider.isSuffocating();
-            this.setSuffocating(!inWarmBlocks && !onWarmStrider);
+            // CraftBukkit start
+            boolean suffocating = !inWarmBlocks && !onWarmStrider;
+            if (suffocating ^ this.isSuffocating()) {
+                if (org.bukkit.craftbukkit.event.CraftEventFactory.callStriderTemperatureChangeEvent(this, suffocating)) {
+                    this.setSuffocating(suffocating);
+                }
+            }
+            // CraftBukkit end
         }
 
         super.tick();
@@ -348,17 +_,17 @@
     }
 
     @Override
-    protected @Nullable SoundEvent getAmbientSound() {
+    public @Nullable SoundEvent getAmbientSound() {
         return !this.isPanicking() && !this.isBeingTempted() ? SoundEvents.STRIDER_AMBIENT : null;
     }
 
     @Override
-    protected SoundEvent getHurtSound(final DamageSource source) {
+    public SoundEvent getHurtSound(final DamageSource source) {
         return SoundEvents.STRIDER_HURT;
     }
 
     @Override
-    protected SoundEvent getDeathSound() {
+    public SoundEvent getDeathSound() {
         return SoundEvents.STRIDER_DEATH;
     }
 
