--- a/net/minecraft/world/entity/projectile/EyeOfEnder.java
+++ b/net/minecraft/world/entity/projectile/EyeOfEnder.java
@@ -26,9 +_,9 @@
     private static final float TOO_FAR_SIGNAL_HEIGHT = 8.0F;
     private static final float TOO_FAR_DISTANCE = 12.0F;
     private static final EntityDataAccessor<ItemStack> DATA_ITEM_STACK = SynchedEntityData.defineId(EyeOfEnder.class, EntityDataSerializers.ITEM_STACK);
-    private @Nullable Vec3 target;
-    private int life;
-    private boolean surviveAfterDeath;
+    public @Nullable Vec3 target;
+    public int life;
+    public boolean surviveAfterDeath;
 
     public EyeOfEnder(final EntityType<? extends EyeOfEnder> type, final Level level) {
         super(type, level);
@@ -73,6 +_,12 @@
     }
 
     public void signalTo(final Vec3 target) {
+        // Paper start - Change EyeOfEnder target without changing other things
+        this.signalTo(target, true);
+    }
+
+    public void signalTo(final Vec3 target, final boolean update) {
+        // Paper end - Change EyeOfEnder target without changing other things
         Vec3 delta = target.subtract(this.position());
         double horizontalDistance = delta.horizontalDistance();
         if (horizontalDistance > 12.0) {
@@ -81,8 +_,10 @@
             this.target = target;
         }
 
+        if (update) { // Paper - Change EyeOfEnder target without changing other things
         this.life = 0;
         this.surviveAfterDeath = this.random.nextInt(5) > 0;
+        } // Paper - Change EyeOfEnder target without changing other things
     }
 
     @Override
@@ -103,7 +_,7 @@
             this.life++;
             if (this.life > 80 && !this.level().isClientSide()) {
                 this.playSound(SoundEvents.ENDER_EYE_DEATH, 1.0F, 1.0F);
-                this.discard();
+                this.discard(this.surviveAfterDeath ? org.bukkit.event.entity.EntityRemoveEvent.Cause.DROP : org.bukkit.event.entity.EntityRemoveEvent.Cause.DESPAWN); // CraftBukkit - add Bukkit remove cause
                 if (this.surviveAfterDeath) {
                     this.level().addFreshEntity(new ItemEntity(this.level(), this.getX(), this.getY(), this.getZ(), this.getItem()));
                 } else {
