--- a/net/minecraft/world/entity/animal/feline/Ocelot.java
+++ b/net/minecraft/world/entity/animal/feline/Ocelot.java
@@ -294,7 +_,7 @@
             }
 
             BlockState state = level.getBlockState(pos.below());
-            if (state.is(Blocks.GRASS_BLOCK) || state.is(BlockTags.LEAVES)) {
+            if (state.is(Blocks.GRASS_BLOCK) || state.isAny(BlockMasks.LEAVES_TAG)) { // Gale - Pre-compute - BlockTags.LEAVES
                 return true;
             }
         }
