--- a/net/minecraft/world/entity/Display.java
+++ b/net/minecraft/world/entity/Display.java
@@ -49,7 +_,7 @@
     private static final EntityDataAccessor<Integer> DATA_TRANSFORMATION_INTERPOLATION_DURATION_ID = SynchedEntityData.defineId(
         Display.class, EntityDataSerializers.INT
     );
-    private static final EntityDataAccessor<Integer> DATA_POS_ROT_INTERPOLATION_DURATION_ID = SynchedEntityData.defineId(
+    public static final EntityDataAccessor<Integer> DATA_POS_ROT_INTERPOLATION_DURATION_ID = SynchedEntityData.defineId(
         Display.class, EntityDataSerializers.INT
     );
     private static final EntityDataAccessor<Vector3fc> DATA_TRANSLATION_ID = SynchedEntityData.defineId(Display.class, EntityDataSerializers.VECTOR3);
@@ -141,7 +_,7 @@
         return false;
     }
 
-    private static Transformation createTransformation(final SynchedEntityData entityData) {
+    public static Transformation createTransformation(final SynchedEntityData entityData) {
         Vector3fc translation = entityData.get(DATA_TRANSLATION_ID);
         Quaternionfc leftRotation = entityData.get(DATA_LEFT_ROTATION_ID);
         Vector3fc scale = entityData.get(DATA_SCALE_ID);
@@ -227,7 +_,7 @@
         this.setBrightnessOverride(input.read("brightness", Brightness.CODEC).orElse(null));
     }
 
-    private void setTransformation(final Transformation transformation) {
+    public void setTransformation(final Transformation transformation) {
         this.entityData.set(DATA_TRANSLATION_ID, transformation.translation());
         this.entityData.set(DATA_LEFT_ROTATION_ID, transformation.leftRotation());
         this.entityData.set(DATA_SCALE_ID, transformation.scale());
@@ -271,19 +_,19 @@
         return this.renderState;
     }
 
-    private void setTransformationInterpolationDuration(final int duration) {
+    public void setTransformationInterpolationDuration(final int duration) {
         this.entityData.set(DATA_TRANSFORMATION_INTERPOLATION_DURATION_ID, duration);
     }
 
-    private int getTransformationInterpolationDuration() {
+    public int getTransformationInterpolationDuration() {
         return this.entityData.get(DATA_TRANSFORMATION_INTERPOLATION_DURATION_ID);
     }
 
-    private void setTransformationInterpolationDelay(final int ticks) {
+    public void setTransformationInterpolationDelay(final int ticks) {
         this.entityData.set(DATA_TRANSFORMATION_INTERPOLATION_START_DELTA_TICKS_ID, ticks, true);
     }
 
-    private int getTransformationInterpolationDelay() {
+    public int getTransformationInterpolationDelay() {
         return this.entityData.get(DATA_TRANSFORMATION_INTERPOLATION_START_DELTA_TICKS_ID);
     }
 
@@ -295,19 +_,19 @@
         return this.entityData.get(DATA_POS_ROT_INTERPOLATION_DURATION_ID);
     }
 
-    private void setBillboardConstraints(final Display.BillboardConstraints constraints) {
+    public void setBillboardConstraints(final Display.BillboardConstraints constraints) {
         this.entityData.set(DATA_BILLBOARD_RENDER_CONSTRAINTS_ID, constraints.getId());
     }
 
-    private Display.BillboardConstraints getBillboardConstraints() {
+    public Display.BillboardConstraints getBillboardConstraints() {
         return Display.BillboardConstraints.BY_ID.apply(this.entityData.get(DATA_BILLBOARD_RENDER_CONSTRAINTS_ID));
     }
 
-    private void setBrightnessOverride(final @Nullable Brightness brightness) {
+    public void setBrightnessOverride(final @Nullable Brightness brightness) {
         this.entityData.set(DATA_BRIGHTNESS_OVERRIDE_ID, brightness != null ? brightness.pack() : -1);
     }
 
-    private @Nullable Brightness getBrightnessOverride() {
+    public @Nullable Brightness getBrightnessOverride() {
         int value = this.entityData.get(DATA_BRIGHTNESS_OVERRIDE_ID);
         return value != -1 ? Brightness.unpack(value) : null;
     }
@@ -316,47 +_,47 @@
         return this.entityData.get(DATA_BRIGHTNESS_OVERRIDE_ID);
     }
 
-    private void setViewRange(final float range) {
+    public void setViewRange(final float range) {
         this.entityData.set(DATA_VIEW_RANGE_ID, range);
     }
 
-    private float getViewRange() {
+    public float getViewRange() {
         return this.entityData.get(DATA_VIEW_RANGE_ID);
     }
 
-    private void setShadowRadius(final float size) {
+    public void setShadowRadius(final float size) {
         this.entityData.set(DATA_SHADOW_RADIUS_ID, size);
     }
 
-    private float getShadowRadius() {
+    public float getShadowRadius() {
         return this.entityData.get(DATA_SHADOW_RADIUS_ID);
     }
 
-    private void setShadowStrength(final float strength) {
+    public void setShadowStrength(final float strength) {
         this.entityData.set(DATA_SHADOW_STRENGTH_ID, strength);
     }
 
-    private float getShadowStrength() {
+    public float getShadowStrength() {
         return this.entityData.get(DATA_SHADOW_STRENGTH_ID);
     }
 
-    private void setWidth(final float width) {
+    public void setWidth(final float width) {
         this.entityData.set(DATA_WIDTH_ID, width);
     }
 
-    private float getWidth() {
+    public float getWidth() {
         return this.entityData.get(DATA_WIDTH_ID);
     }
 
-    private void setHeight(final float width) {
+    public void setHeight(final float width) {
         this.entityData.set(DATA_HEIGHT_ID, width);
     }
 
-    private int getGlowColorOverride() {
+    public int getGlowColorOverride() {
         return this.entityData.get(DATA_GLOW_COLOR_OVERRIDE_ID);
     }
 
-    private void setGlowColorOverride(final int value) {
+    public void setGlowColorOverride(final int value) {
         this.entityData.set(DATA_GLOW_COLOR_OVERRIDE_ID, value);
     }
 
@@ -373,7 +_,7 @@
         return result;
     }
 
-    private float getHeight() {
+    public float getHeight() {
         return this.entityData.get(DATA_HEIGHT_ID);
     }
 
@@ -483,11 +_,11 @@
             }
         }
 
-        private BlockState getBlockState() {
+        public BlockState getBlockState() {
             return this.entityData.get(DATA_BLOCK_STATE_ID);
         }
 
-        private void setBlockState(final BlockState blockState) {
+        public void setBlockState(final BlockState blockState) {
             this.entityData.set(DATA_BLOCK_STATE_ID, blockState);
         }
 
@@ -579,19 +_,19 @@
             }
         }
 
-        private ItemStack getItemStack() {
+        public ItemStack getItemStack() {
             return this.entityData.get(DATA_ITEM_STACK_ID);
         }
 
-        private void setItemStack(final ItemStack item) {
+        public void setItemStack(final ItemStack item) {
             this.entityData.set(DATA_ITEM_STACK_ID, item);
         }
 
-        private void setItemTransform(final ItemDisplayContext transform) {
+        public void setItemTransform(final ItemDisplayContext transform) {
             this.entityData.set(DATA_ITEM_DISPLAY_ID, transform.getId());
         }
 
-        private ItemDisplayContext getItemTransform() {
+        public ItemDisplayContext getItemTransform() {
             return ItemDisplayContext.BY_ID.apply(this.entityData.get(DATA_ITEM_DISPLAY_ID));
         }
 
@@ -673,8 +_,8 @@
         public static final int INITIAL_BACKGROUND = 1073741824;
         private static final int INITIAL_LINE_WIDTH = 200;
         private static final EntityDataAccessor<Component> DATA_TEXT_ID = SynchedEntityData.defineId(Display.TextDisplay.class, EntityDataSerializers.COMPONENT);
-        private static final EntityDataAccessor<Integer> DATA_LINE_WIDTH_ID = SynchedEntityData.defineId(Display.TextDisplay.class, EntityDataSerializers.INT);
-        private static final EntityDataAccessor<Integer> DATA_BACKGROUND_COLOR_ID = SynchedEntityData.defineId(
+        public static final EntityDataAccessor<Integer> DATA_LINE_WIDTH_ID = SynchedEntityData.defineId(Display.TextDisplay.class, EntityDataSerializers.INT);
+        public static final EntityDataAccessor<Integer> DATA_BACKGROUND_COLOR_ID = SynchedEntityData.defineId(
             Display.TextDisplay.class, EntityDataSerializers.INT
         );
         private static final EntityDataAccessor<Byte> DATA_TEXT_OPACITY_ID = SynchedEntityData.defineId(Display.TextDisplay.class, EntityDataSerializers.BYTE);
@@ -707,15 +_,15 @@
             }
         }
 
-        private Component getText() {
+        public Component getText() {
             return this.entityData.get(DATA_TEXT_ID);
         }
 
-        private void setText(final Component text) {
+        public void setText(final Component text) {
             this.entityData.set(DATA_TEXT_ID, text);
         }
 
-        private int getLineWidth() {
+        public int getLineWidth() {
             return this.entityData.get(DATA_LINE_WIDTH_ID);
         }
 
@@ -723,15 +_,15 @@
             this.entityData.set(DATA_LINE_WIDTH_ID, width);
         }
 
-        private byte getTextOpacity() {
+        public byte getTextOpacity() {
             return this.entityData.get(DATA_TEXT_OPACITY_ID);
         }
 
-        private void setTextOpacity(final byte opacity) {
+        public void setTextOpacity(final byte opacity) {
             this.entityData.set(DATA_TEXT_OPACITY_ID, opacity);
         }
 
-        private int getBackgroundColor() {
+        public int getBackgroundColor() {
             return this.entityData.get(DATA_BACKGROUND_COLOR_ID);
         }
 
@@ -739,11 +_,11 @@
             this.entityData.set(DATA_BACKGROUND_COLOR_ID, color);
         }
 
-        private byte getFlags() {
+        public byte getFlags() {
             return this.entityData.get(DATA_STYLE_FLAGS_ID);
         }
 
-        private void setFlags(final byte flags) {
+        public void setFlags(final byte flags) {
             this.entityData.set(DATA_STYLE_FLAGS_ID, flags);
         }
 
