--- a/net/minecraft/world/entity/Interaction.java
+++ b/net/minecraft/world/entity/Interaction.java
@@ -34,8 +_,8 @@
     private static final float DEFAULT_WIDTH = 1.0F;
     private static final float DEFAULT_HEIGHT = 1.0F;
     private static final boolean DEFAULT_RESPONSE = false;
-    private Interaction.@Nullable PlayerAction attack;
-    private Interaction.@Nullable PlayerAction interaction;
+    public Interaction.@Nullable PlayerAction attack;
+    public Interaction.@Nullable PlayerAction interaction;
 
     public Interaction(final EntityType<?> type, final Level level) {
         super(type, level);
@@ -99,9 +_,16 @@
     @Override
     public boolean skipAttackInteraction(final Entity source) {
         if (source instanceof Player player) {
+            // CraftBukkit start
+            DamageSource damageSource = player.damageSources().generic().eventEntityDamager(source);
+            org.bukkit.event.entity.EntityDamageEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callNonLivingEntityDamageEvent(this, damageSource, 1.0F, false);
+            if (event.isCancelled()) {
+                return true;
+            }
+            // CraftBukkit end
             this.attack = new Interaction.PlayerAction(player.getUUID(), this.level().getGameTime());
             if (player instanceof ServerPlayer serverPlayer) {
-                CriteriaTriggers.PLAYER_HURT_ENTITY.trigger(serverPlayer, this, player.damageSources().generic(), 1.0F, 1.0F, false);
+                CriteriaTriggers.PLAYER_HURT_ENTITY.trigger(serverPlayer, this, damageSource, 1.0F, (float) event.getFinalDamage(), false); // CraftBukkit
             }
 
             return !this.getResponse();
@@ -139,27 +_,27 @@
         return this.interaction != null ? this.level().getPlayerByUUID(this.interaction.player()) : null;
     }
 
-    private void setWidth(final float width) {
+    public void setWidth(final float width) {
         this.entityData.set(DATA_WIDTH_ID, width);
     }
 
-    private float getWidth() {
+    public float getWidth() {
         return this.entityData.get(DATA_WIDTH_ID);
     }
 
-    private void setHeight(final float width) {
+    public void setHeight(final float width) {
         this.entityData.set(DATA_HEIGHT_ID, width);
     }
 
-    private float getHeight() {
+    public float getHeight() {
         return this.entityData.get(DATA_HEIGHT_ID);
     }
 
-    private void setResponse(final boolean response) {
+    public void setResponse(final boolean response) {
         this.entityData.set(DATA_RESPONSE_ID, response);
     }
 
-    private boolean getResponse() {
+    public boolean getResponse() {
         return this.entityData.get(DATA_RESPONSE_ID);
     }
 
@@ -177,7 +_,7 @@
         return this.getDimensions().makeBoundingBox(position);
     }
 
-    private record PlayerAction(UUID player, long timestamp) {
+    public record PlayerAction(UUID player, long timestamp) {
         public static final Codec<Interaction.PlayerAction> CODEC = RecordCodecBuilder.create(
             i -> i.group(
                     UUIDUtil.CODEC.fieldOf("player").forGetter(Interaction.PlayerAction::player),
