--- a/net/minecraft/world/entity/OminousItemSpawner.java
+++ b/net/minecraft/world/entity/OminousItemSpawner.java
@@ -28,7 +_,7 @@
     private static final String TAG_ITEM = "item";
     private static final EntityDataAccessor<ItemStack> DATA_ITEM = SynchedEntityData.defineId(OminousItemSpawner.class, EntityDataSerializers.ITEM_STACK);
     public static final int TICKS_BEFORE_ABOUT_TO_SPAWN_SOUND = 36;
-    private long spawnItemAfterTicks;
+    public long spawnItemAfterTicks;
 
     public OminousItemSpawner(final EntityType<? extends OminousItemSpawner> type, final Level level) {
         super(type, level);
@@ -78,7 +_,7 @@
                     spawnedEntity = this.spawnProjectile(level, projectileItem, item);
                 } else {
                     spawnedEntity = new ItemEntity(level, this.getX(), this.getY(), this.getZ(), item);
-                    level.addFreshEntity(spawnedEntity);
+                    level.addFreshEntity(spawnedEntity, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.OMINOUS_ITEM_SPAWNER); // Paper - fixes and addition to spawn reason API
                 }
 
                 level.levelEvent(LevelEvent.PARTICLES_TRIAL_SPAWNER_SPAWN_ITEM, this.blockPosition(), 1);
@@ -92,7 +_,7 @@
         ProjectileItem.DispenseConfig dispenseConfig = projectileItem.createDispenseConfig();
         dispenseConfig.overrideDispenseEvent().ifPresent(event -> level.levelEvent(event, this.blockPosition(), 0));
         Direction direction = Direction.DOWN;
-        Projectile projectile = Projectile.spawnProjectileUsingShoot(
+        Projectile projectile = Projectile.spawnProjectileUsingShootDelayed( // Paper - fixes and addition to spawn reason API
             projectileItem.asProjectile(level, this.position(), item, direction),
             level,
             item,
@@ -101,7 +_,7 @@
             direction.getStepZ(),
             dispenseConfig.power(),
             dispenseConfig.uncertainty()
-        );
+        ).spawn(org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.OMINOUS_ITEM_SPAWNER); // Paper - fixes and addition to spawn reason API
         projectile.setOwner(this);
         return projectile;
     }
@@ -180,7 +_,7 @@
         return this.getEntityData().get(DATA_ITEM);
     }
 
-    private void setItem(final ItemStack itemStack) {
+    public void setItem(final ItemStack itemStack) {
         this.getEntityData().set(DATA_ITEM, itemStack);
     }
 
