--- a/net/minecraft/world/item/context/UseOnContext.java
+++ b/net/minecraft/world/item/context/UseOnContext.java
@@ -21,7 +_,7 @@
         this(player.level(), player, hand, player.getItemInHand(hand), hitResult);
     }
 
-    protected UseOnContext(
+    public UseOnContext(
         final Level level, final @Nullable Player player, final InteractionHand hand, final ItemStack itemStack, final BlockHitResult hitResult
     ) {
         this.player = player;
