--- a/net/minecraft/util/datafix/fixes/BlockStateData.java
+++ b/net/minecraft/util/datafix/fixes/BlockStateData.java
@@ -683,7 +_,7 @@
         return new Dynamic<>(JavaOps.INSTANCE, Map.of("Name", name, "Properties", properties)).convert(NbtOps.INSTANCE);
     }
 
-    private static void register(final int id, final Dynamic<?> tag, final Dynamic<?>... legacy) {
+    public static void register(final int id, final Dynamic<?> tag, final Dynamic<?>... legacy) {
         MAP[id] = tag;
         int blockId = id >> 4;
         if (BLOCK_DEFAULTS[blockId] == null) {
