--- a/net/minecraft/world/entity/animal/squid/GlowSquid.java
+++ b/net/minecraft/world/entity/animal/squid/GlowSquid.java
@@ -53,17 +_,17 @@
     }
 
     @Override
-    protected SoundEvent getAmbientSound() {
+    public SoundEvent getAmbientSound() {
         return SoundEvents.GLOW_SQUID_AMBIENT;
     }
 
     @Override
-    protected SoundEvent getHurtSound(final DamageSource source) {
+    public SoundEvent getHurtSound(final DamageSource source) {
         return SoundEvents.GLOW_SQUID_HURT;
     }
 
     @Override
-    protected SoundEvent getDeathSound() {
+    public SoundEvent getDeathSound() {
         return SoundEvents.GLOW_SQUID_DEATH;
     }
 
@@ -100,7 +_,7 @@
         return hurt;
     }
 
-    private void setDarkTicks(final int ticks) {
+    public void setDarkTicks(final int ticks) {
         this.entityData.set(DATA_DARK_TICKS_REMAINING, ticks);
     }
 
