--- a/net/minecraft/world/entity/decoration/Mannequin.java
+++ b/net/minecraft/world/entity/decoration/Mannequin.java
@@ -33,8 +_,8 @@
     private static final EntityDataAccessor<Optional<Component>> DATA_DESCRIPTION = SynchedEntityData.defineId(
         Mannequin.class, EntityDataSerializers.OPTIONAL_COMPONENT
     );
-    private static final byte ALL_LAYERS = (byte)Arrays.stream(PlayerModelPart.values()).mapToInt(PlayerModelPart::getMask).reduce(0, (a, b) -> a | b);
-    private static final Set<Pose> VALID_POSES = Set.of(Pose.STANDING, Pose.CROUCHING, Pose.SWIMMING, Pose.FALL_FLYING, Pose.SLEEPING);
+    public static final byte ALL_LAYERS = (byte)Arrays.stream(PlayerModelPart.values()).mapToInt(PlayerModelPart::getMask).reduce(0, (a, b) -> a | b);
+    public static final Set<Pose> VALID_POSES = Set.of(Pose.STANDING, Pose.CROUCHING, Pose.SWIMMING, Pose.FALL_FLYING, Pose.SLEEPING);
     public static final Codec<Pose> POSE_CODEC = Pose.CODEC
         .validate(pose -> VALID_POSES.contains(pose) ? DataResult.success(pose) : DataResult.error(() -> "Invalid pose: " + pose.getSerializedName()));
     private static final Codec<Byte> LAYERS_CODEC = PlayerModelPart.CODEC
@@ -44,7 +_,7 @@
             mask -> Arrays.stream(PlayerModelPart.values()).filter(part -> (mask & part.getMask()) == 0).toList()
         );
     public static final ResolvableProfile DEFAULT_PROFILE = ResolvableProfile.Static.EMPTY;
-    private static final Component DEFAULT_DESCRIPTION = Component.translatable("entity.minecraft.mannequin.label");
+    public static final Component DEFAULT_DESCRIPTION = Component.translatable("entity.minecraft.mannequin.label");
     protected static EntityType.EntityFactory<Mannequin> constructor = Mannequin::new;
     private static final String PROFILE_FIELD = "profile";
     private static final String HIDDEN_LAYERS_FIELD = "hidden_layers";
@@ -82,28 +_,28 @@
         return this.entityData.get(DATA_PROFILE);
     }
 
-    private void setProfile(final ResolvableProfile profile) {
+    public void setProfile(final ResolvableProfile profile) {
         this.entityData.set(DATA_PROFILE, profile);
     }
 
-    private boolean getImmovable() {
+    public boolean getImmovable() {
         return this.entityData.get(DATA_IMMOVABLE);
     }
 
-    private void setImmovable(final boolean immovable) {
+    public void setImmovable(final boolean immovable) {
         this.entityData.set(DATA_IMMOVABLE, immovable);
     }
 
-    protected @Nullable Component getDescription() {
+    public @Nullable Component getDescription() {
         return this.entityData.get(DATA_DESCRIPTION).orElse(null);
     }
 
-    private void setDescription(final Component description) {
+    public void setDescription(final Component description) {
         this.description = description;
         this.updateDescription();
     }
 
-    private void setHideDescription(final boolean hideDescription) {
+    public void setHideDescription(final boolean hideDescription) {
         this.hideDescription = hideDescription;
         this.updateDescription();
     }
