--- a/net/minecraft/world/level/block/EnchantingTableBlock.java
+++ b/net/minecraft/world/level/block/EnchantingTableBlock.java
@@ -101,7 +_,7 @@
     }
 
     @Override
-    protected @Nullable MenuProvider getMenuProvider(final BlockState state, final Level level, final BlockPos pos) {
+    public @Nullable MenuProvider getMenuProvider(final BlockState state, final Level level, final BlockPos pos) {
         if (level.getBlockEntity(pos) instanceof EnchantingTableBlockEntity enchantingTable) {
             Component title = enchantingTable.getDisplayName();
             return new SimpleMenuProvider(
