--- a/net/minecraft/world/item/ItemStackLinkedSet.java
+++ b/net/minecraft/world/item/ItemStackLinkedSet.java
@@ -6,7 +_,7 @@
 import org.jspecify.annotations.Nullable;
 
 public class ItemStackLinkedSet {
-    private static final Strategy<? super ItemStack> TYPE_AND_TAG = new Strategy<ItemStack>() {
+    public static final Strategy<? super ItemStack> TYPE_AND_TAG = new Strategy<ItemStack>() {
         @Override
         public int hashCode(final @Nullable ItemStack item) {
             return ItemStack.hashItemAndComponents(item);
