--- a/net/minecraft/world/level/pathfinder/Path.java
+++ b/net/minecraft/world/level/pathfinder/Path.java
@@ -13,7 +_,7 @@
 
 public final class Path {
     public static final StreamCodec<FriendlyByteBuf, Path> STREAM_CODEC = StreamCodec.of((output, value) -> value.writeToStream(output), Path::createFromStream);
-    private final List<Node> nodes;
+    public final List<Node> nodes;
     private Path.@Nullable DebugData debugData;
     private int nextNodeIndex;
     private final BlockPos target;
