--- a/net/minecraft/world/level/levelgen/structure/templatesystem/loader/TemplateSource.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/loader/TemplateSource.java
@@ -57,7 +_,7 @@
         }
     }
 
-    private static CompoundTag readStructure(final InputStream input) throws IOException {
+    public static CompoundTag readStructure(final InputStream input) throws IOException {
         return NbtIo.readCompressed(input, NbtAccounter.unlimitedHeap());
     }
 
@@ -68,7 +_,7 @@
         }
     }
 
-    private StructureTemplate readStructure(final CompoundTag tag) {
+    public StructureTemplate readStructure(final CompoundTag tag) {
         StructureTemplate structureTemplate = new StructureTemplate();
         int version = NbtUtils.getDataVersion(tag, 500);
         structureTemplate.load(this.blockLookup, DataFixTypes.STRUCTURE.updateToCurrentVersion(this.fixerUpper, tag, version));
