--- a/net/minecraft/nbt/ListTag.java
+++ b/net/minecraft/nbt/ListTag.java
@@ -147,7 +_,7 @@
         this(new ArrayList<>());
     }
 
-    ListTag(final List<Tag> list) {
+    public ListTag(final List<Tag> list) {
         this.list = list;
     }
 
@@ -190,7 +_,7 @@
     }
 
     @VisibleForTesting
-    byte identifyRawElementType() {
+    public byte identifyRawElementType() {
         byte homogenousType = Tag.TAG_END;
 
         for (Tag element : this.list) {
