--- a/net/minecraft/resources/Identifier.java
+++ b/net/minecraft/resources/Identifier.java
@@ -25,6 +25,10 @@
     public static final String ALLOWED_NAMESPACE_CHARACTERS = "[a-z0-9_.-]";
     private final String namespace;
     private final String path;
+    // Gale start - Cache identifier toString and hash
+    private String identifierStr;
+    private int hash;
+    // Gale end - Cache identifier toString and hash

     private Identifier(final String namespace, final String path) {
         assert isValidNamespace(namespace);
@@ -123,7 +127,14 @@

     @Override
     public String toString() {
-        return this.namespace + ":" + this.path;
+        // Gale start - Cache identifier toString and hash
+        String identifierStr = this.identifierStr;
+        if (identifierStr == null) {
+            identifierStr = this.namespace + ":" + this.path;
+            this.identifierStr = identifierStr;
+        }
+        return identifierStr;
+        // Gale end - Cache identifier toString and hash
     }

     @Override
@@ -133,7 +144,14 @@

     @Override
     public int hashCode() {
-        return 31 * this.namespace.hashCode() + this.path.hashCode();
+        // Gale start - Cache identifier toString and hash
+        int hash = this.hash;
+        if (hash == 0) {
+            hash = 31 * this.namespace.hashCode() + this.path.hashCode();
+            this.hash = hash;
+        }
+        return hash;
+        // Gale end - Cache identifier toString and hash
     }

     @Override
