--- a/net/minecraft/world/item/crafting/Ingredient.java
+++ b/net/minecraft/world/item/crafting/Ingredient.java
@@ -22,7 +_,7 @@
 import net.minecraft.world.item.crafting.display.SlotDisplay;
 import net.minecraft.world.level.ItemLike;
 
-public final class Ingredient implements Predicate<ItemStack>, StackedContents.IngredientInfo<Holder<Item>> {
+public final class Ingredient implements Predicate<ItemStack>, StackedContents.IngredientInfo<io.papermc.paper.inventory.recipe.ItemOrExact> { // Paper - Improve exact choice recipe ingredients
     public static final StreamCodec<RegistryFriendlyByteBuf, Ingredient> CONTENTS_STREAM_CODEC = ByteBufCodecs.holderSet(Registries.ITEM)
         .map(Ingredient::new, i -> i.values);
     public static final StreamCodec<RegistryFriendlyByteBuf, Optional<Ingredient>> OPTIONAL_CONTENTS_STREAM_CODEC = ByteBufCodecs.holderSet(Registries.ITEM)
@@ -32,7 +_,28 @@
         );
     public static final Codec<HolderSet<Item>> NON_AIR_HOLDER_SET_CODEC = HolderSetCodec.create(Registries.ITEM, Item.CODEC, false);
     public static final Codec<Ingredient> CODEC = ExtraCodecs.nonEmptyHolderSet(NON_AIR_HOLDER_SET_CODEC).xmap(Ingredient::new, i -> i.values);
-    private final HolderSet<Item> values;
+    public final HolderSet<Item> values;
+    // CraftBukkit start
+    private java.util.@org.jspecify.annotations.Nullable Set<ItemStack> itemStacks; // Paper - Improve exact choice recipe ingredients
+
+    public boolean isExact() {
+        return this.itemStacks != null;
+    }
+
+    public java.util.@org.jspecify.annotations.Nullable Set<ItemStack> itemStacks() { // Paper - Improve exact choice recipe ingredients
+        return this.itemStacks;
+    }
+
+    public static Ingredient ofStacks(java.util.List<ItemStack> stacks) {
+        Ingredient ingredient = Ingredient.of(stacks.stream().map(ItemStack::getItem));
+        // Paper start - Improve exact choice recipe ingredients
+        ingredient.itemStacks = net.minecraft.world.item.ItemStackLinkedSet.createTypeAndComponentsSet();
+        ingredient.itemStacks.addAll(stacks);
+        ingredient.itemStacks = java.util.Collections.unmodifiableSet(ingredient.itemStacks);
+        // Paper end - Improve exact choice recipe ingredients
+        return ingredient;
+    }
+    // CraftBukkit end
 
     private Ingredient(final HolderSet<Item> values) {
         values.unwrap().ifRight(directValues -> {
@@ -62,17 +_,29 @@
 
     @Override
     public boolean test(final ItemStack input) {
+        // CraftBukkit start
+        if (this.isExact()) {
+            return this.itemStacks.contains(input); // Paper - Improve exact choice recipe ingredients (hashing FTW!)
+        }
+        // CraftBukkit end
         return input.is(this.values);
     }
 
+    // Paper start - Improve exact choice recipe ingredients
     @Override
-    public boolean acceptsItem(final Holder<Item> item) {
-        return this.values.contains(item);
+    public boolean acceptsItem(final io.papermc.paper.inventory.recipe.ItemOrExact itemOrExact) {
+        return switch (itemOrExact) {
+            case io.papermc.paper.inventory.recipe.ItemOrExact.Item(final Holder<Item> item) ->
+                !this.isExact() && this.values.contains(item);
+            case io.papermc.paper.inventory.recipe.ItemOrExact.Exact(final ItemStack exact) ->
+                this.isExact() && this.itemStacks.contains(exact);
+        };
+        // Paper end - Improve exact choice recipe ingredients
     }
 
     @Override
     public boolean equals(final Object o) {
-        return o instanceof Ingredient other && Objects.equals(this.values, other.values);
+        return o instanceof Ingredient other && Objects.equals(this.values, other.values) && Objects.equals(this.itemStacks, other.itemStacks); // CraftBukkit
     }
 
     @Override
@@ -97,6 +_,11 @@
     }
 
     public SlotDisplay display() {
+        // Paper start - show exact ingredients in recipe book
+        if (this.isExact()) {
+            return new SlotDisplay.Composite(this.itemStacks().stream().map(ItemStackTemplate::fromNonEmptyStack).<SlotDisplay>map(SlotDisplay.ItemStackSlotDisplay::new).toList());
+        }
+        // Paper end - show exact ingredients in recipe book
         return (SlotDisplay)this.values
             .unwrap()
             .map(SlotDisplay.TagSlotDisplay::new, l -> new SlotDisplay.Composite(l.stream().map(Ingredient::displayForSingleItem).toList()));
