--- a/net/minecraft/world/entity/ai/attributes/AttributeSupplier.java
+++ b/net/minecraft/world/entity/ai/attributes/AttributeSupplier.java
@@ -14,7 +_,7 @@
         this.instances = instances;
     }
 
-    private AttributeInstance getAttributeInstance(final Holder<Attribute> attribute) {
+    public AttributeInstance getAttributeInstance(final Holder<Attribute> attribute) {
         AttributeInstance instance = this.instances.get(attribute);
         if (instance == null) {
             throw new IllegalArgumentException("Can't find attribute " + attribute.getRegisteredName());
