--- a/net/minecraft/world/entity/ai/attributes/Attribute.java
+++ b/net/minecraft/world/entity/ai/attributes/Attribute.java
@@ -15,7 +_,7 @@
     private final double defaultValue;
     private boolean syncable;
     private final String descriptionId;
-    private Attribute.Sentiment sentiment = Attribute.Sentiment.POSITIVE;
+    public Attribute.Sentiment sentiment = Attribute.Sentiment.POSITIVE;
 
     protected Attribute(final String descriptionId, final double defaultValue) {
         this.defaultValue = defaultValue;
