--- a/net/minecraft/world/entity/projectile/arrow/Arrow.java
+++ b/net/minecraft/world/entity/projectile/arrow/Arrow.java
@@ -19,7 +_,7 @@
 
 public class Arrow extends AbstractArrow {
     private static final int EXPOSED_POTION_DECAY_TIME = 600;
-    private static final int NO_EFFECT_COLOR = -1;
+    public static final int NO_EFFECT_COLOR = -1;
     private static final EntityDataAccessor<Integer> ID_EFFECT_COLOR = SynchedEntityData.defineId(Arrow.class, EntityDataSerializers.INT);
     private static final byte EVENT_POTION_PUFF = 0;
 
@@ -37,7 +_,7 @@
         this.updateColor();
     }
 
-    private PotionContents getPotionContents() {
+    public PotionContents getPotionContents() {
         return this.getPickupItemStackOrigin().getOrDefault(DataComponents.POTION_CONTENTS, PotionContents.EMPTY);
     }
 
@@ -45,13 +_,13 @@
         return this.getPickupItemStackOrigin().getOrDefault(DataComponents.POTION_DURATION_SCALE, 1.0F);
     }
 
-    private void setPotionContents(final PotionContents potionContents) {
+    public void setPotionContents(final PotionContents potionContents) {
         this.getPickupItemStackOrigin().set(DataComponents.POTION_CONTENTS, potionContents);
         this.updateColor();
     }
 
     @Override
-    protected void setPickupItemStack(final ItemStack itemStack) {
+    public void setPickupItemStack(final ItemStack itemStack) {
         super.setPickupItemStack(itemStack);
         this.updateColor();
     }
@@ -71,6 +_,16 @@
         entityData.define(ID_EFFECT_COLOR, -1);
     }
 
+    // Paper start - EAR 2
+    @Override
+    public void inactiveTick() {
+        if (this.isInGround()) {
+            this.life++;
+        }
+        super.inactiveTick();
+    }
+    // Paper end
+
     @Override
     public void tick() {
         super.tick();
@@ -116,7 +_,7 @@
         Entity effectSource = this.getEffectSource();
         PotionContents potionContents = this.getPotionContents();
         float durationScale = this.getPotionDurationScale();
-        potionContents.forEachEffect(effect -> mob.addEffect(effect, effectSource), durationScale);
+        potionContents.forEachEffect(effect -> mob.addEffect(effect, effectSource, org.bukkit.event.entity.EntityPotionEffectEvent.Cause.ARROW), durationScale); // CraftBukkit
     }
 
     @Override
