--- a/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplateManager.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplateManager.java
@@ -45,8 +_,8 @@
     private static final FileToIdConverter WORLD_TEXT_STRUCTURE_LISTER = new FileToIdConverter("structure", ".snbt");
     private static final FileToIdConverter RESOURCE_STRUCTURE_LISTER = new FileToIdConverter("structure", ".nbt");
     public static final FileToIdConverter RESOURCE_TEXT_STRUCTURE_LISTER = new FileToIdConverter("structure", ".snbt");
-    private final Map<Identifier, Optional<StructureTemplate>> structureRepository = new ConcurrentHashMap<>();
-    private final ResourceManagerTemplateSource resourceManagerSource;
+    public final Map<Identifier, Optional<StructureTemplate>> structureRepository = new ConcurrentHashMap<>();
+    public final ResourceManagerTemplateSource resourceManagerSource;
     private final List<TemplateSource> sources;
     private final TemplatePathFactory worldTemplates;
     private final @Nullable TemplatePathFactory testTemplates;
@@ -99,7 +_,7 @@
         return this.sources.stream().flatMap(TemplateSource::list).distinct();
     }
 
-    private Optional<StructureTemplate> tryLoad(final Identifier id) {
+    public Optional<StructureTemplate> tryLoad(final Identifier id) {
         for (TemplateSource source : this.sources) {
             try {
                 Optional<StructureTemplate> loaded = source.load(id);
